### PR TITLE
Stop using by-id paths for devices getting encrypted [DualPass]

### PR DESCRIPTION
**Pull request by vermashi**

### CODE REVIEW

File: VMEncryption/main/DiskUtil.py [hunk 1]
```diff
@@ -635,6 +635,8 @@ def query_dev_sdx_path_by_uuid(self, uuid):
     def query_dev_id_path_by_sdx_path(self, sdx_path):
         """
         return /dev/disk/by-id that maps to the sdx_path, otherwise return the original path
+        Update: now we have realised that by-id is not a good way to refer to devices (they can change on reallocations or resizes).
+        Try not to use this- use get_stable_path_from_sdx instead
```
Comment (ejarvi) [inline VMEncryption/main/DiskUtil.py]: is this dead code now? if so remove

File: VMEncryption/main/DiskUtil.py [hunk 2]
```diff
@@ -643,31 +645,26 @@ def query_dev_id_path_by_sdx_path(self, sdx_path):
...
+            return azure_name_table[sdx_realpath]
+
+        # Then try matching a uuid symlink. Those are probably the best
+        for disk_by_uuid in os.listdir(CommonVariables.disk_by_uuid_root):
+            disk_by_uuid_path = os.path.join(CommonVariables.disk_by_uuid_root, disk_by_uuid)
+
+            if os.path.realpath(disk_by_uuid_path) == sdx_realpath:
+                return disk_by_uuid_path
+
+        # Found nothing very persistent. Just return the original sdx path.
```
Comment (ejarvi) [inline VMEncryption/main/DiskUtil.py]: add error or warning if get_persistent_path failed to get a persistent path

File: VMEncryption/main/handle.py [hunk 3]
```diff
@@ -130,7 +130,8 @@ def disable_encryption():
             if crypt_item.dev_path.startswith("/dev/sd"):
                 logger.log('Updating crypt item entry to use mapper name')
                 logger.log('Device name before update: {0}'.format(crypt_item.dev_path))
-                crypt_item.dev_path = disk_util.query_dev_id_path_by_sdx_path(crypt_item.dev_path)
+                # crypt_item.dev_path = disk_util.query_dev_id_path_by_sdx_path(crypt_item.dev_path)
```
Comment (ejarvi) [inline VMEncryption/main/handle.py]: no comment needed, version control will save this (same as other line changes below)

**Comment by mayank88mahajan** [comment]
@mayank88mahajan

**Pull request merged**